pkgs/servers/code-server/default.nix

```diff
@@ -133,79 +140,68 @@ in stdenv.mkDerivation rec {
+    jq --slurp '.[0] * .[1]' "./lib/vscode/product.json" <(
       cat << EOF
     {
       "builtInExtensions": []
     }
     EOF
-    ) | sponge vendor/modules/code-oss-dev/product.json
+    ) | sponge ./lib/vscode/product.json
 
     # disable automatic updates
     sed -i '/update.mode/,/\}/{s/default:.*/default: "none",/g}' \
-      vendor/modules/code-oss-dev/src/vs/platform/update/common/update.config.contribution.ts
-
-    # put ripgrep binary into bin, so postinstall does not try to download it
-    find -name vscode-ripgrep -type d \
-      -execdir mkdir -p {}/bin \; \
-      -execdir ln -s ${ripgrep}/bin/rg {}/bin/rg \;
-
-    # Playwright is only needed for tests, we can disable it for builds.
-    # There's an environment variable to disable downloads, but the package makes a breaking call to
-    # sw_vers before that variable is checked.
-    patch -p1 -i ${./playwright.patch}
+      lib/vscode/src/vs/platform/update/common/update.config.contribution.ts
 
     # Patch out remote download of nodejs from build script
     patch -p1 -i ${./remove-node-download.patch}
 
-    # Replicate install vscode dependencies without running script for all vscode packages
-    # that require patching for postinstall scripts to succeed
-    find ./vendor/modules/code-oss-dev -path "*node_modules" -prune -o \
-      -path "./*/*/*/*/*" -name "yarn.lock" -printf "%h\n" | \
+    # Fetch packages for vscode
+    find ./lib/vscode -name "yarn.lock" -printf "%h\n" | \
         xargs -I {} yarn --cwd {} \
-          --frozen-lockfile --offline --ignore-scripts --ignore-engines
-
+          --frozen-lockfile --ignore-scripts --ignore-engines
 
     # patch shebangs of everything to allow binary packages to build
     patchShebangs .
 
-    ${patchEsbuild "./vendor/modules/code-oss-dev/build" "0.12.6"}
-    ${patchEsbuild "./vendor/modules/code-oss-dev/extensions" "0.11.23"}
-  '' + lib.optionalString stdenv.isDarwin ''
+    ${patchEsbuild "./lib/vscode/build" "0.12.6"}
+    ${patchEsbuild "./lib/vscode/extensions" "0.11.23"}
+    '' + lib.optionalString stdenv.isDarwin ''
     # use prebuilt binary for @parcel/watcher, which requires macOS SDK 10.13+
     # (see issue #101229)
-    pushd ./vendor/modules/code-oss-dev/remote/node_modules/@parcel/watcher
+    pushd ./lib/vscode/remote/node_modules/@parcel/watcher
     mkdir -p ./build/Release
     mv ./prebuilds/darwin-x64/node.napi.glibc.node ./build/Release/watcher.node
     jq "del(.scripts) | .gypfile = false" ./package.json | sponge ./package.json
     popd
   '' + ''
-    # rebuild binaries, we use npm here, as yarn does not provide an alternative
-    # that would not attempt to try to reinstall everything and break our
-    # patching attempts
-    npm rebuild --prefix vendor/modules/code-oss-dev --update-binary
+
+    # put ripgrep binary into bin, so postinstall does not try to download it
+    find -name ripgrep -type d \
```

Comment:
Can we use relative paths instead of using find?

Reply:
this statement could be replaced but would increase maintenance complexity as the paths changed include hashes

Reply:
We could cd with wildcards into the directory. Not sure if thats better or not.